File: swift/array-hashmap/int-array-duplicates.swift
import Foundation

func containsDuplicate(_ nums: [Int]) -> Bool {
    var seen = Set<Int>()
    for num in nums {
        if seen.contains(num) {
            return true
        }
        seen.insert(num)
    }
    return false
}

print(containsDuplicate([1, 2, 3, 4]))  // false
print(containsDuplicate([1, 2, 3, 4, 1]))  // true

File: kotlin/array-hashmap/int-array-duplicates.kt
fun containsDuplicate(nums: IntArray): Boolean {
    val unique = mutableSetOf<Int>()
    for (num in nums) {
        if (unique.contains(num)) return true
        unique.add(num)
    }
    return false
}

fun main() {
    println(containsDuplicate(intArrayOf(1, 2, 3, 4)))      // false
    println(containsDuplicate(intArrayOf(1, 2, 3, 4, 1)))   // true
}
